File: Visitor Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Visitor Pattern in Java");
        System.out.println("-----------------------");
        System.out.println();
        
        Element elementA = new ConcreteElementA();
        Element elementB = new ConcreteElementB();
    
        Visitor visitorA = new ConcreteVisitorA();
        Visitor visitorB = new ConcreteVisitorB();
    
        elementA.accept(visitorA);
        elementA.accept(visitorB);
    
        elementB.accept(visitorA);
        elementB.accept(visitorB);
    }
}

File: Visitor Pattern/1. Minimal/Python/main.py
from concrete_element_a import ConcreteElementA
from concrete_element_b import ConcreteElementB
from concrete_visitor_a import ConcreteVisitorA
from concrete_visitor_b import ConcreteVisitorB

def main():
    print("Visitor Pattern in Python")
    print("-------------------------")
    print()
    
    elementA = ConcreteElementA()
    elementB = ConcreteElementB()

    visitorA = ConcreteVisitorA()
    visitorB = ConcreteVisitorB()

    elementA.accept(visitorA)
    elementA.accept(visitorB)

    elementB.accept(visitorA)
    elementB.accept(visitorB)

if __name__ == "__main__":
    main()
